h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstPivot.java

```diff
@@ -44,90 +45,95 @@ public ValFrame exec(Val[] args) {
     // This is the sort then MRTask method.
     // Create the target Frame
     // Now sort on the index key, result is that unique keys will be localized
-    Frame fr2 = fr.sort(new int[]{0});
+    Frame fr2 = fr.sort(new int[]{indexIdx});
     final long[] classes = new VecUtils.CollectDomain().doAll(fr.vec(colIdx)).domain();
-    final int nClass = fr.vec(colIdx).isNumeric() ? classes.length : fr.vec(colIdx).domain().length;
-    String[] header = (String[]) ArrayUtils.addAll(new String[]{index}, fr.vec(colIdx).domain());
-    Frame initialPassPreSort = new pivotTask(fr2,index,column,value)
+    final int nClass = (fr.vec(colIdx).isNumeric() || fr.vec(colIdx).isTime()) ? classes.length : fr.vec(colIdx).domain().length;
+    String[] header = null;
+    if (fr.vec(colIdx).isNumeric()) {
+      header = (String[]) ArrayUtils.addAll(new String[]{index}, Arrays.toString(classes).split("[\\[\\]]")[1].split(", "));
+    } else if (fr.vec(colIdx).isTime()) {
+      header = new String[nClass];
+      for (int i=0;i<nClass;i++) header[i] = (new DateTime(classes[i], DateTimeZone.UTC)).toString();
```

Comment:
Ah, I see. We can assume UTC then. Just have to let users know. Or give them the chance to convert.